src/check_datapackage/check.py

```diff
@@ -52,15 +52,34 @@ def __init__(
         # TODO: Switch to using `explain()` once implemented
         errors: list[str] = _map(
             issues,
-            lambda issue: f"- Property `{issue.jsonpath}`: {issue.message}\n",
+            explain,
         )
         message: str = (
-            "There were some issues found in your `datapackage.json`:\n\n"
+            f"There were {len(errors)} issues found in your `datapackage.json`:\n\n"
             + "\n".join(errors)
         )
         super().__init__(message)
 
 
+def explain(issue: Issue) -> str:
+    """Explain the issue in a human-readable format.
+
+    Args:
+        issue: The `Issue` object to explain.
+
+    Returns:
+        A human-readable explanation of the issue.
+    """
+    property_name = issue.jsonpath.split(".")[-1]
+    return (  # noqa: F401
+        f"At package.{issue.jsonpath[2:]}:\n"  # indexing to remove '$.'
+        "|\n"
+        f"| {property_name}: {issue.instance!r}\n"
+        f"| {' ' * len(property_name)}  {'^' * len(str(issue.instance))}\n"
+        f"{issue.message}\n"
```

Comment:
I think we might need to have some custom messages, as some of the messages are not helpful (e.g. the regex ones).

But that can be another PR :relaxed: